go/vt/vtgate/planbuilder/horizon_planning.go

```diff
@@ -837,11 +838,13 @@ func createMemorySortPlanOnAggregation(plan *orderedAggregate, orderExprs []abst
 		if !found {
 			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "expected to find the order by expression (%s) in orderedAggregate", sqlparser.String(order.Inner))
 		}
+		// TODO(king-11) not sure whether below will work or not.
```

Comment:
you should add a unit test and an end-to-end test if you are uncertain

Reply:
Sure next task to do in my list for this PR.

Reply:
I am not able to find where should I add in tests for this particular change. I added in tests for the `routeSort` change though.